### PR TITLE
[PFX-684] - Externalize Prompts for Preset use

### PR DESCRIPTION
**Pull request by mmason2-godaddy**
## Summary
Externalize certain _plugin-specific_ prompts to allow them to be programmatically invoked in the `presetPrompt` lifecycle. 

- `package.json` exports
  - [nextjs plugin](https://github.com/godaddy/gasket/blob/PFX-684/packages/gasket-plugin-nextjs/package.json#L5-L8)
  - [typescript plugin](https://github.com/godaddy/gasket/blob/PFX-684/packages/gasket-plugin-typescript/package.json#L5-L8)
- Prompt file
  - [nextjs plugin](https://github.com/godaddy/gasket/blob/PFX-684/packages/gasket-plugin-nextjs/package.json#L5-L8)
  - [typescript plugin](https://github.com/godaddy/gasket/blob/PFX-684/packages/gasket-plugin-typescript/lib/prompt.js)
- Presets updated
  - [nextjs preset](https://github.com/godaddy/gasket/blob/PFX-684/packages/gasket-preset-nextjs/lib/preset-prompt.js)
  - [api preset](https://github.com/godaddy/gasket/blob/PFX-684/packages/gasket-preset-api/lib/preset-prompt.js)
- Added `fastify` to dependencies and type files(necessary - the preset needs the plugins loaded even if they're conditional)
- **NOTE:** _context control structures_ need to be contained in the exported functions in order to be externalized correctly.

## Changelog
- Externalize certain _plugin-specific_ prompts to allow them to be programmatically invoked in the `presetPrompt` lifecycle.

## Test Plan
Tests & types updated

### CODE REVIEW

File: packages/gasket-preset-api/lib/preset-config.js [hunk 1]
```diff
@@ -14,6 +13,10 @@ import pluginLint from '@gasket/plugin-lint';
 export default async function presetConfig(gasket, context) {
   let typescriptPlugin;
   const testPlugins = [];
+  // TODO: test canary publish without fastify dependency
```
Comment (agerard-godaddy) [inline packages/gasket-preset-api/lib/preset-config.js:16]: Still need this comment?

File: packages/gasket-plugin-nextjs/lib/prompt.js [hunk 2]
```diff
@@ -1,71 +1,87 @@
...
-    const { addSitemap } = await prompt([
-      {
-        name: 'addSitemap',
-        message: 'Do you want to add a sitemap?',
-        type: 'confirm',
-        default: false
-      }
-    ]);
+/** @type {import('@gasket/core').HookHandler<'prompt'>} */
+async function prompt(gasket, context, { prompt }) {
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-nextjs/lib/prompt.js:72]: ❤️

**Review by agerard-godaddy — APPROVED** [review]
This turned out great! 🥇 Much more DRY yet flexible.

File: packages/gasket-preset-api/lib/preset-config.js (hunk 1 above)
Comment (mmason2-godaddy) [reply packages/gasket-preset-api/lib/preset-config.js:16]: Good catch will remove 👍

**Review by jpina1-godaddy — APPROVED** [review]

**Pull request merged**